### DISCUSSION TITLE
Control Microphone gain

### DISCUSSION THREAD

**Discussion by aviporje** [pschatzmann/arduino-audio-tools#1595] [Q&A] [answered]
Respected Sir Thanks for Audio Tool Library, Its Perfect
i using VBAN Protocol to transmeet Audio, its working Perfect but i having small issues with microphone
its noise like 4 stroke engine, so i think i have to control microphone gain
but dont know how? I using ESP32 A1S AC101 and VBAN Audio Kit please help me to set microphone gain level
Thanks

Device Description
ESP32 A1S AC101

**Comment by pschatzmann** [comment, marked as answer]
There is a [chapter in the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/Volume-Control) that describes how to control the volume.

If you use the [AudioDriver](https://github.com/pschatzmann/arduino-audio-driver) library you can also use the functionality provided by the hardware by calling setInputVolume

**Reply by aviporje** [reply]
Thank you sir
